src/main/java/org/jenkinsci/plugins/workflow/steps/StepDescriptor.java

```diff
@@ -305,6 +306,27 @@ public boolean apply(StepDescriptor i) {
         return r;
     }
 
+    /** Return true if we can easily create a nice String for user display from the object */
+    static boolean isAbleToUseToStringForDisplay(@CheckForNull Object o) {
+        return (o != null && (o instanceof Comparable || o instanceof CharSequence)); // Covers our base types
```

Comment:
@stephenc Actually we want implementers to override *this* method very rarely.  

What they should be overriding is argumentsToString and doing their own explicit logic -- in which case they won't be invoking it at all.